backend/clickhouse/traces.go

```diff
@@ -122,3 +125,67 @@ func convertLinks(traceRow *TraceRow) (clickhouse.ArraySet, clickhouse.ArraySet,
 	}
 	return traceIDs, spanIDs, states, attrs
 }
+
+func (client *Client) ReadTraces(ctx context.Context, projectID int, params modelInputs.TracesParamsInput) ([]*modelInputs.Trace, error) {
+	sb := sqlbuilder.NewSelectBuilder()
+	var err error
+	var args []interface{}
+
+	sb.From("traces").
+		Select("Timestamp, UUID, TraceId, SpanId, ParentSpanId, ProjectId, SecureSessionId, TraceState, SpanName, SpanKind, Duration, ServiceName, ServiceVersion, TraceAttributes, StatusCode, StatusMessage").
+		Where(sb.Equal("ProjectId", projectID))
+
+	// Very basic filtering for now. Will add better support + pagination later.
+	queryParams := queryparser.Parse(params.Query)
+	for key, value := range queryParams.Attributes {
+		sb.Where(sb.Equal(key, value[0]))
+	}
+
+	sb.OrderBy("Timestamp DESC").Limit(100)
+	sql, args := sb.Build()
```

Comment:
there's also a `BuildWithFlavor` where you can specify `Clickhouse` - not sure if it makes a difference in this case